nilearn/_utils/estimator_checks.py

```diff
@@ -678,7 +675,7 @@ def fit_estimator(estimator: BaseEstimator) -> BaseEstimator:
         data[2, 2, 2, y] = 2
         X = Nifti1Image(data, np.eye(4))
 
-        return estimator.fit(X, y)
+        return X, y
```

Comment:
Instead of each of these return statements, single statement outside of if conditions could avoid repetitions.

Reply:
yeah but sometimes in several cases the 'nature / type' of the second variable will be very different: in some cases it is an actually "y" required for fitting, sometimes is it is a design matrix...

so I'd rather not.

in fact when I try to do it the mypy starts complaining because of a given variable is suddenly getting an unexpected type.